src/esp/nav/PathFinder.cpp

```diff
@@ -34,6 +35,41 @@ namespace Cr = Corrade;
 namespace esp {
 namespace nav {
 
+struct MultiGoalShortestPath::Impl {
+  Impl() = default;
+  ~Impl() = default;
+
+  void setRequestedEnds(const std::vector<vec3f>& newEnds) {
+    endRefs.clear();
+    pathEnds.clear();
+    requestedEnds = newEnds;
+
+    minTheoreticalDist.assign(newEnds.size(), 0);
+  }
+
+  const std::vector<vec3f>& getRequestedEnds() const { return requestedEnds; }
+
+  std::vector<vec3f> requestedEnds;
+
+  std::vector<dtPolyRef> endRefs;
+  std::vector<vec3f> pathEnds;
+
+  std::vector<float> minTheoreticalDist;
+  vec3f prevRequestedStart = vec3f::Zero();
+};
+
+MultiGoalShortestPath::MultiGoalShortestPath()
+    : pimpl_{spimpl::make_unique_impl<Impl>()} {};
+
+void MultiGoalShortestPath::setRequestedEnds(
+    const std::vector<vec3f>& newEnds) {
+  pimpl_->setRequestedEnds(newEnds);
+}
+
+const std::vector<vec3f>& MultiGoalShortestPath::getRequestedEnds() const {
+  return pimpl_->getRequestedEnds();
+}
```

Comment:
Probably more of a style thing, but since `Impl` is a struct with everything public, why not just this, and getting rid of all methods on the struct? You're accessing the members directly elsewhere anyway. (This is how I do PIMPLs myself.)

```suggestion
void MultiGoalShortestPath::setRequestedEnds(
    const std::vector<vec3f>& newEnds) {
    pimpl_->endRefs.clear();
    pimpl_->pathEnds.clear();
    pimpl_->requestedEnds = newEnds;

    pimpl_->minTheoreticalDist.assign(newEnds.size(), 0);
}

const std::vector<vec3f>& MultiGoalShortestPath::getRequestedEnds() const {
  return pimpl_->requestedEnds;
}
```

Reply:
Yeah, this makes more sense.  I am using PIMPL kinda strangely here as I wanted to hide the thing that needs a setter/getter, but not the other things.